net/host.go

```diff
@@ -308,86 +334,113 @@ func (h *host) addrInfo() peer.AddrInfo {
 	}
 }
 
-func (h *host) writeToStream(s libp2pnetwork.Stream, msg Message) error {
+func writeStreamMessage(s io.Writer, msg Message, peerID peer.ID) error {
```

Comment:
why change stream to `io.Writer` and add the additional peerID param?

Reply:
The peer ID was only used in a log message. I changed the type to make the function unit testable.